hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/CompactStreamSerializer.java

```diff
@@ -260,14 +232,28 @@ private CompactSerializableRegistration getOrCreateRegistration(String typeName)
         });
     }
 
+    private GenericRecord readGenericRecord(BufferObjectDataInput input, Schema schema, boolean schemaIncludedInBinary) {
+        DefaultCompactReader reader = new DefaultCompactReader(this, input, schema, null, schemaIncludedInBinary);
+        Collection<FieldDescriptor> fields = schema.getFields();
+        DeserializedSchemaBoundGenericRecordBuilder builder = new DeserializedSchemaBoundGenericRecordBuilder(schema);
+        for (FieldDescriptor fieldDescriptor : fields) {
+            String fieldName = fieldDescriptor.getFieldName();
+            FieldKind fieldKind = fieldDescriptor.getKind();
+            builder.write(fieldName, reader.readAny(fieldName), fieldKind);
+        }
+        return builder.build();
```

Comment:
There seem to be some unnecessary checks in these two calls. Maybe we can put a TODO comment here to improve things in the future